Realm/ObjectStore/object_store_exceptions.hpp

```diff
@@ -24,42 +24,78 @@
 #include <string>
 
 namespace realm {
+    class Property;
 
     class ObjectStoreException : public std::exception {
     public:
         enum class Kind {
```

Comment:
What I'm wondering, though, is how big the difference actually is in amount of boilerplate code, considering the type safety and locality of the alternative. But let's leave it for now and take it up when we start solidifying the API. :)